pkgs/by-name/ma/maltego/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchzip
+, jre
+, giflib
+, gawk
+, runtimeShell
+}:
+
+stdenv.mkDerivation rec {
+  pname = "maltego";
+  version = "4.6.0";
+  src = fetchzip {
+    url = "https://downloads.maltego.com/maltego-v4/linux/Maltego.v${version}.linux.zip";
+    hash = "sha256-q+1RYToZtBxAIDSiUWf3i/3GBBDwh6NWteHiK4VM1HY=";
+  };
+
+  preConfigure = ''
+      for awks in bin/maltego; do
+          substituteInPlace $awks \
+                --replace /usr/bin/awk ${gawk}/bin/awk
+      done
+  '';
+
+  buildInputs = [ jre giflib ];
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mkdir -p "$out/share"
+    cp -aR . "$out/share/${pname}/"
+    cat > "$out/bin/${pname}" << EOF
```

Comment:
What if you use `makeWrapper` here? I believe you can use something like this:

```
makeWrapper $out/share/${pname}/bin/${pname} $out/bin/${pname} \
  --set JAVA_HOME=${jre}
  --prefix PATH : ${lib.makeBinPath [ jre ]}
```

Reply:
I can try it but my question is: why does it work when I build the package and generate the binary by
```nix
nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}'
```
while if you use `nixpkgs-preview` no? What is the correct command to use `nixpkgs-preview` to test the package?

Reply:
This observation is unrelated to the issue indicated above. Also, testing through `nixpkgs-review` sometimes bring an unexpected issue. I am going to test it in another way.

Reply:
I still get the same issue if putting to build locally, unfortunately:

```
> nix build .#maltego --impure
warning: Ignoring setting 'auto-allocate-uids' because experimental feature 'auto-allocate-uids' is not enabled
warning: Ignoring setting 'impure-env' because experimental feature 'configurable-impure-env' is not enabled
> ./result/bin/maltego
java is /nix/store/hxbq5qcdbqyy86zc5k2sz8ln2af4vmw0-openjdk-19.0.2+7/bin/java
found java executable in PATH
pwd: file:/nix/store/w5dn7c6zq6cqfh4zlbj9mclafvm80ich-maltego-4.6.0/share/maltego/maltego-ui/modules/ext/Java_Config_App.jar
install conf: /nix/store/w5dn7c6zq6cqfh4zlbj9mclafvm80ich-maltego-4.6.0/share/maltego/etc/maltego.conf
install version: v4.6.0
user conf: /home/vlore/.maltego/v4.6.0/etc/maltego.conf
JAVA_HOME: /nix/store/hxbq5qcdbqyy86zc5k2sz8ln2af4vmw0-openjdk-19.0.2+7
current java: /nix/store/hxbq5qcdbqyy86zc5k2sz8ln2af4vmw0-openjdk-19.0.2+7/lib/openjdk
Exception in thread "main" java.lang.NullPointerException: Cannot read the array length because "<local8>" is null
	at com.paterva.maltego.java.config.jre.unix.UnixDetectJava.getExistingSearchPaths(UnixDetectJava.java:81)
	at com.paterva.maltego.java.config.jre.unix.UnixDetectJava.getPathsToTry(UnixDetectJava.java:69)
	at com.paterva.maltego.java.config.jre.DetectJavas.detectJavas(DetectJavas.java:32)
	at com.paterva.maltego.java.config.jre.JavaUtils.detectJavas(JavaUtils.java:29)
	at com.paterva.maltego.java.config.app.SilentDetector.go(SilentDetector.java:23)
	at com.paterva.maltego.java.config.app.JavaOptionsApp.main(JavaOptionsApp.java:47)
```

Reply:
I will create an isolated environment to build it and I check what I get. I will keep you updated. Thanks

Reply:
I generated an environment by `nix-shell` by the following `shell.nix`:
```nix
with import <nixpkgs> {};
stdenv.mkDerivation rec {
  name = "env";
  buildInputs = [
    git
  ];
}
```
Then I run:
```nix
nix-shell
nix-build -E 'with import <nixpkgs> {}; callPackage ./package.nix {}'
cd result/bin
./maltego
```
and it works...
![image](https://github.com/NixOS/nixpkgs/assets/83867734/563c6fc0-6b79-499a-84c4-9e3f42fcaafe)

Reply:
@WolfangAukang I performed some tests with other Nix users. We are using Linux as host and it seems to work correctly. Are you using a different system? For example darwin-based?

Reply:
Got it, then my case is isolated. I will review the code in this case.

Reply:
Btw @D3vil0p3r, were you able to test the `makeWrapper`? I am throwing this recommendation because it looks cleaner, but it if does not work, then it is not a problem.

Reply:
Implemented. It works correctly. I noted also that, despite the `src` downloads the `linux` maltego package from official website, it seems to work also for darwin (MacOS Apple M1) by building the submitted `package.nix` and running its binary.